core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -194,9 +194,18 @@ protected void doStop() {
                 if (holderToNotify != null) {
                     // callback that an exception happened, but on a different thread since we don't
                     // want handlers to worry about stack overflows
-                    threadPool.generic().execute(new Runnable() {
+                    threadPool.generic().execute(new AbstractRunnable() {
                         @Override
-                        public void run() {
+                        public void onRejection(Throwable t) {
+                            // if we get rejected during node shutdown we don't wanna bubble it up
+                            logger.debug("failed to notify response handler on rejection", t);
```

Comment:
@bleskes I really cannot see a scenario in which we would have a lot of noise that you are describing. Just to make sure I didn't miss something obvious I set these loggers to error level and ran all our integration tests in core 3 times with error logging. I saw this message logged only ones. Considering how rare this condition is getting triggered and how potentially useful this debug message can be for threading issues I think we should leave it at the debug level. The trace level in TransportService is way too chatty to be useful.